code/modules/uplink/uplink_item.dm

```diff
@@ -101,7 +101,10 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
 		for(var/obj/item/I in B)
 			U.purchase_log += "<big>\icon[I]</big>"
 	else
-		U.purchase_log += "<big>\icon[A]</big>"
+		// Don't add /obj/item/stack/telecrystal to the purchase_log since
+		// it's just used to buy more items (including itself!)
+		if(!istype(src, /datum/uplink_item/device_tools/telecrystal))
```

Comment:
Easy enough to do. My only argument would be that this _is_ probably the only thing we want to suppress from the purchase log, since it's essentially currency (and usually currency is an acceptable special-case). Checking on the datum is more expandable but I can't think of something it would immediately apply too (other then this, obviously).

But that's more of an intellectual argument. If you insist I'll change it.